wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Content/MessagePresenter.swift

```diff
@@ -204,28 +207,46 @@ final class MessagePresenter: NSObject {
 
     func viewController(forImageMessagePreview message: ZMConversationMessage, actionResponder delegate: MessageActionResponder) -> UIViewController? {
         guard Message.isImage(message),
-            message.imageMessageData != nil else {
-                return nil
+              message.imageMessageData != nil else {
+            return nil
         }
 
         return imagesViewController(for: message, actionResponder: delegate, isPreviewing: true)
     }
 
     // MARK: - Pass
 
-    func createAddPassesViewController(fileMessageData: ZMFileMessageData) -> PKAddPassesViewController? {
-        guard let fileURL = fileMessageData.fileURL,
-            let passData = try? Data.init(contentsOf: fileURL) else {
-                return nil
+    @MainActor
+    func openPassesViewController(fileMessageData: ZMFileMessageData) async {
+        guard PKAddPassesViewController.canAddPasses() else { return }
+
+        guard let viewController = try? await makePassesViewController(fileMessageData: fileMessageData) else {
+            return // ignore error
         }
 
-        guard let pass = try? PKPass.init(data: passData) else { return nil }
+        targetViewController?.present(viewController, animated: true)
+    }
 
-        if PKAddPassesViewController.canAddPasses() {
-            return PKAddPassesViewController(pass: pass)
-        } else {
-            return nil
+    @MainActor
+    func makePassesViewController(fileMessageData: ZMFileMessageData) async throws -> PKAddPassesViewController? {
```

Comment:
I understand, that `PKAddPassesViewController.init` returns an optional if  ["there was a problem initializing the object"](https://developer.apple.com/documentation/passkit/pkaddpassesviewcontroller/1619251-init).
However, since we return an error for `missingFileURL` we should add an error case for when the initialisation of the `PKAddPassesViewController` failed.

Reply:
Could be a good idea, so you mean we should change the method return type to `PKAddPassesViewController` without the optional then?

Reply:
Exactly, sorry for being unprecise.

Reply:
Done